### PR TITLE
[SD-3351] Legal Archive improvements

### PR DESCRIPTION
**Pull request by sivakuna-aap**
After merging this we need to remove the articles as the structure of groups has been changed.

### CODE REVIEW

**Comment by akintolga** [comment]
:+1:

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -190,10 +190,10 @@
          *
          * @param {string} _id Item _id.
          * @param {boolean} read_only
-         * @param {string} itemType
+         * @param {string} repo - repository where an item whose identifier is _id can be found.
          */
-        this.open = function openAuthoring(_id, read_only, itemType) {
-            if (angular.isDefined(itemType) && itemType === 'legal_archive') {
+        this.open = function openAuthoring(_id, read_only, repo) {
+            if (repo && repo === 'legal_archive') {
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Just saying `repo === 'legal_archive'` is already enough here.

File: server/apps/legal_archive/service.py [hunk 2]
```diff
@@ -18,22 +18,73 @@
...
+                    self._change_location_of_items_in_package(doc)
+
+    def on_replace(self, document, original):
+        """
+        Overriding to replace the location of each item in the package to legal archive instead of archive,
+        if doc is a pacakge.
+        """
+
+        super().on_replace(document, original)
+        if ITEM_TYPE in document and document[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE:
```
Comment (plamut) [inline server/apps/legal_archive/service.py]: This condition can be simplified a bit with the `get()` method:
```py
if document.get(ITEM_TYPE) == CONTENT_TYPE.COMPOSITE:
```

**Comment by plamut** [comment]
There are a couple of deprecated `self.assertEquals()` calls in the tests, but generally it looks good to me.

:+1:

**Comment by mdhaman** [comment]
:+1:

**Comment by sivakuna-aap** [comment]
Closing this as this is dependent on [PR#1221](https://github.com/superdesk/superdesk/pull/1221)

**Comment by actionless** [comment]
> sivakuna-aap added the manual deployment needed 

what does this tag means? is there anything need to be done from bamboo side?

**Comment by amagdas** [comment]
@actionless it means that data cleanup is needed for all the content

**Pull request closed without merging**